app/models/king.rb

```diff
@@ -1,2 +1,11 @@
 class King < Piece
+	def valid_move?(new_rank, new_file)
+		rank_diff = (new_rank - rank).abs
+		file_diff = (new_file - file).abs
+		if (rank_diff <= 1) && (file_diff <= 1) && !is_move_obstructed?(new_rank, new_file) && !is_move_malformed?(new_rank, new_file)
```

Comment:
`is_move_obstructed?` is not designed to handle malformed moves and will generally be the most expensive procedure that needs to be called as part of `valid_move?`, so I would use the following order instead:

- Call `is_move_malformed?`
- Perform any type-specific validations
- Call `is_move_obstructed?` if necessary (it won't be for the Knight model)